app/components/billing-status-badge.ts

```diff
@@ -22,10 +22,48 @@ export default class BillingStatusDisplayComponent extends Component<Signature>
     return this.currentUser?.activeDiscountForYearlyPlan || null;
   }
 
+  get activeDiscountForYearlyPlanExcludingStage2CompletionDiscount(): PromotionalDiscountModel | null {
+    return this.currentUser?.activeDiscountForYearlyPlanExcludingStage2CompletionDiscount || null;
+  }
+
+  get badgeType(): 'vip' | 'member' | 'old-discounts' | 'all-discounts' | 'free-weeks-left' | 'upgrade' {
+    if (this.shouldShowVipBadge) {
+      return 'vip';
+    }
+
+    if (this.shouldShowMemberBadge) {
+      return 'member';
+    }
+
+    if (this.activeDiscountForYearlyPlan) {
```

Comment:
Can simplify this overall: 

```
if (this.activeDiscountForYearlyPlan && this.canSeeDiscountCountdown) {
  if (!this.activeDiscountForYearlyPlan.isFromStage2Completion) {
    return "discount-timer"
  }
  
  // Stage 2 completion discounts are hidden behind a feature flag 
  if (this.activeDiscountForYearlyPlan.isFromStage2Completion && this.canSeeStage2CompletionDiscount) {
    return "discount-timer"
  }
}

Reply:
Oh wait, I now see why this can't be done and you need `activeDiscountForYearlyExcluding...`.

Reply:
Okay, revised: 

```
if (this.activeDiscountForYearlyPlan && this.canSeeDiscountCountdown) {
  // Stage 2 completion discounts are hidden behind a feature flag 
  if (this.activeDiscountForYearlyPlan.isFromStage2Completion && this.canSeeStage2CompletionDiscount) {
    return "discount-timer"
  }
  
  if (this.activeDiscountForYearlyPlanExcludingStage2CompletionDiscount) {
    return "discount-timer"
  }
}
```